types/type_map.go

```diff
@@ -99,3 +103,16 @@ func (t typeMap) Validate() error {
 	}
 	return t.value.Validate()
 }
+
+func genName(baseName string, key bool) string {
+	postfix := "value"
+	if key {
+		postfix = "key"
+	}
+	digest := md5.New()
+	n, err := digest.Write([]byte(baseName + postfix))
```

Comment:
1) There can be collisions in MD5
2) You have a bug -  `value5` should be like `valued131dd02c5e6eec4 693d9a0698aff95c` by your logic
3) Counter is faster and readable. See previous point.